net/pfSense-pkg-frr/files/usr/local/pkg/frr/inc/frr_zebra.inc

```diff
@@ -620,6 +620,21 @@ function frr_zebra_generate_acceptfilters() {
 	return $noaccept;
 }
 
+/* The following function checks for the presence of and writes the contents of "/var/etc/frr/$module.conf" into the $module . "running" field within the config.xml file  */
+function write_frr_running_config($module) {
+	global $config, $frr_config_base;
+
+	$moduleRunningFile = "{$frr_config_base}/{$module}.conf";
+
+	if ( file_exists( $moduleRunningFile ) && ( filesize( $moduleRunningFile ) > 0 ) ) {
+		$moduleRunning = fopen( "$moduleRunningFile", "r" );
+		$config['installedpackages']['frrglobalraw']['config'][0][$module . "running"] = base64_encode( fread( $moduleRunning, filesize($moduleRunningFile) ));
+ 	} else {
+		$config['installedpackages']['frrglobalraw']['config'][0][$module . "running"] = base64_encode("!!!!! {$module}.conf does not exist or is empty.");
+	}
+	write_config( $desc = gettext("FRR: Wrote {$module}.conf startup-config to pfSense config file."), $backup = false, $write_config_only = true );
```

Comment:
Why is this setting variables by name here in the function call? Seems like it could be problematic. If this was copied directly over from quagga I suspect this may also be why on quagga I see lots of config writes which update the running config with a broken description of `(system): 1`.

This will need tested and maybe fixed both here and in quagga.

Reply:
When I wrote that function for Quagga, the idea of the variable being part of the function call was to reduce/simplify the code, allowing the function to be used to write the config for any module.

Reply:
fixed & rebased